packages/noya-renderer/src/components/DistanceLabelAndPath.tsx

```diff
@@ -0,0 +1,323 @@
+import {
+  Polyline,
+  Text,
+  useReactCanvasKit,
+  Rect as RCKRect,
+  useFontManager,
+  useColorFill,
+} from 'noya-react-canvaskit';
+import React, { useMemo } from 'react';
+import { createBounds, Point, Rect, distance } from 'noya-geometry';
+
+type DistanceMeasurementProps = {
+  distance: number;
+  bounds: Point;
+};
+
+function DistanceMeasurement({ distance, bounds }: DistanceMeasurementProps) {
+  const { CanvasKit } = useReactCanvasKit();
+  const fontManager = useFontManager();
+  const paragraph = useMemo(() => {
+    const paragraphStyle = new CanvasKit.ParagraphStyle({
+      textStyle: {
+        color: CanvasKit.parseColorString('rgb(255,255,225)'),
+        fontSize: 11,
+        fontFamilies: ['Roboto'],
+        letterSpacing: 0.2,
+      },
+    });
+
+    const builder = CanvasKit.ParagraphBuilder.Make(
+      paragraphStyle,
+      fontManager,
+    );
+    builder.addText(distance.toString());
+
+    const paragraph = builder.build();
+    paragraph.layout(10000);
+
+    return paragraph;
+  }, [CanvasKit, fontManager, distance]);
+
+  const labelRect = useMemo(
+    () =>
+      CanvasKit.XYWHRect(
+        bounds.x,
+        bounds.y,
+        paragraph.getMinIntrinsicWidth(),
+        paragraph.getHeight(),
+      ),
+    [CanvasKit, paragraph, bounds],
+  );
+
+  const bgRect = useMemo(
+    () =>
+      CanvasKit.XYWHRect(
+        bounds.x - 2,
+        bounds.y - 1,
+        paragraph.getMinIntrinsicWidth() + 5,
+        paragraph.getHeight() + 2,
+      ),
+    [CanvasKit, paragraph, bounds],
+  );
+
+  const backgroundFill = useColorFill('rgb(43, 92, 207)');
+
+  return (
+    <>
+      <RCKRect rect={bgRect} paint={backgroundFill} />
+      <Text rect={labelRect} paragraph={paragraph} />
+    </>
+  );
+}
+
+function MeasureAndExtensionPaths({
+  selectedLayer,
+  highlightedLayer,
+}: {
+  selectedLayer: Rect;
+  highlightedLayer: Rect;
+}) {
+  const highlightedBounds = createBounds(highlightedLayer);
+  const selectedBounds = createBounds(selectedLayer);
+
+  let implicitLines: Point[][] = [];
+  let explicitLines: Point[][] = [];
+  let distanceMeasurements: DistanceMeasurementProps[] = [];
```

Comment:
I found that sometimes the text labels appear on the opposite side of the explicit lines, when the explicit lines are drawn inward. If there's a simple fix, we should do it as part of this issue, otherwise we can come back to it.

![Screen Shot 2021-05-14 at 1 55 35 PM](https://user-images.githubusercontent.com/1198882/118332065-594df800-b4be-11eb-9094-db5e6e783598.png)

Reply:
Midpoint calculation was incorrect, issue has been fixed. 
<img width="358" alt="Screen Shot 2021-05-14 at 5 07 27 PM" src="https://user-images.githubusercontent.com/43770359/118342193-2c5a0f00-b4d7-11eb-97b0-6fba239f7617.png">